lib/chef/policy_builder/policyfile.rb

```diff
@@ -263,6 +263,19 @@ def run_list_expansion_ish
       def apply_policyfile_attributes
         node.attributes.role_default = policy["default_attributes"]
         node.attributes.role_override = policy["override_attributes"]
+        policy_group = (defined?(node.policy_group) && node.policy_group) || \
```

Comment:
you probably don't need the defensive coding around `defined?(node.policy_group)` here?

Reply:
Yeah, that was because that cookbook had to run on pre-policyfile versions of chef :)

Reply:
In fact all of this logic shouldn't be needed. We already unify all the data here https://github.com/chef/chef/blob/599f0dfec0c8c2b0d6d6eaf2a594abeebf40ff66/lib/chef/policy_builder/policyfile.rb#L386-L395

Reply:
Oh gotcha thanks, didn't spot that at all!